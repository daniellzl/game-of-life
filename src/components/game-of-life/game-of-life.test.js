import { shallow } from "enzyme";
import GameOfLife from "./game-of-life.component";

const BOARD = [
  [1, 1, 0, 0, 0],
  [1, 0, 1, 0, 1],
  [1, 1, 0, 0, 0],
  [0, 0, 1, 0, 1],
  [0, 0, 0, 0, 1],
];

it("doesn't explode", () => {
  expect(shallow(<GameOfLife initialBoard={BOARD} />)).toMatchSnapshot();
});

// it("can start a game", () => {
//   const wrapper = shallow(<GameOfLife />);
//   wrapper.find('[id="start-game"]').simulate("click");
//   // expect(wrapper.state()).toEqual({ count: 2 });
//   console.log(wrapper.state());
// });
