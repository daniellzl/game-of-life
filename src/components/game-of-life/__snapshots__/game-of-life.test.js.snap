// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't explode 1`] = `
<div
  className="gameOfLife"
>
  <div
    className="title"
  >
    Game of Life
  </div>
  <div
    className="generations"
  >
    Generations: 
    0
  </div>
  <div
    className="board"
  >
    <div
      key="row0"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col0"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col1"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col2"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col3"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col4"
      />
    </div>
    <div
      key="row1"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col0"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col1"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col2"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col3"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col4"
      />
    </div>
    <div
      key="row2"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col0"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col1"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col2"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col3"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col4"
      />
    </div>
    <div
      key="row3"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col0"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col1"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col2"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col3"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col4"
      />
    </div>
    <div
      key="row4"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col0"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col1"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col2"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={false}
        isLiveGame={false}
        key="col3"
      />
      <GridCell
        handleCellClick={[Function]}
        isAlive={true}
        isLiveGame={false}
        key="col4"
      />
    </div>
  </div>
  <div
    className="buttons"
  >
    <button
      className="nes-btn is-success button"
      disabled={false}
      onClick={[Function]}
    >
      START
    </button>
    <button
      className="nes-btn is-disabled button"
      disabled={true}
      onClick={[Function]}
    >
      STOP
    </button>
    <button
      className="nes-btn is-warning button"
      onClick={[Function]}
    >
      CLEAR
    </button>
    <button
      className="nes-btn is-primary button"
      onClick={[Function]}
    >
      RESET
    </button>
  </div>
</div>
`;
